mifospay/src/main/java/org/mifos/mobilewallet/mifospay/history/presenter/HistoryPresenter.java

```diff
@@ -90,6 +95,37 @@ public void handleTransactionClick(int transactionIndex) {
         mHistoryView.showTransactionDetailDialog(transactionIndex, accountNumber);
     }
 
+    public List<Transaction> getFilteredTransactions(List<Transaction> transactions,
+                                                     List<CheckBoxStatus> statusList) {
+        ArrayList<Transaction> filteredTransactions = new ArrayList<>();
+        List<CheckBoxStatus> CheckBoxStatusList = getCheckedStatusList(statusList);
+        for (Transaction transaction : transactions) {
```

Comment:
avoid nested for loops especially more so because number of transactions can get very large.

Reply:
Thanks for the feedback, I agree the nested for loops look ugly but my thought process while doing this was that for a particular filtering process the number of checkboxes or filter parameters are constant so the runtime scales linearly in the number of transactions making its runtime O(n).